### DISCUSSION TITLE
online map not showing when built from source code.

### DISCUSSION THREAD

**Discussion by muihlinn** [foldynl/QLog#200] [Q&A]
I have a laptop with latest QLog from ppa, works without issues AFAIK. 

On the desktop computer I've built QLog from sources as indicated in the readme and online map doesn't show anywhere.  The only difference I can tell is that binary was built with QT5 and the one from sources with QT6.

Both machines run ubuntu, 20.4 (laptop) and 22.4 (desktop).

Any hints or something I've missed out?

**Comment by foldynl** [comment]
That's strange.

I have installed a fresh Ubuntu 22.04 (Basic Installation). Then I installed all components from Readme for QT6. It means 
`sudo apt-get -y install libhamlib-dev build-essential pkg-config qt6-base-dev qtkeychain-qt6-dev qt6-webengine-dev libqt6charts6-dev libqt6serialport6-dev libqt6webenginecore6-bin libqt6svg6-dev libgl-dev`

Then I called 

`qmake6 QLog.pro`

and 

`make`.

And a result is that the QLog contains the maps.

Please, could you verify this?

1) In the Main Menu -> Help, please, verify whether QLog under QT6 is really compiled against QT6 - the dialog shows the QT version
![image](https://user-images.githubusercontent.com/942769/235887973-dd469023-4ba2-454f-9091-0258340aec3f.png)

2) if you have really installed libqt6webenginecore6-bin.

when you run QLog in the console. In the case of the QT6 application, additional information is listed. Can you send them to me? In my case it is:

```
Warning: Ignoring WAYLAND_DISPLAY on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
12:09:16.036 [INFO    ] [0x7f403b8bd480]                              [qt.webenginecontext] => 

GLImplementation: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.3
Using Default SG Backend: yes
Using Software Dynamic GL: no
Using Angle: no

Init Parameters:
  *  application-name QLog 
  *  browser-subprocess-path /usr/lib/qt6/libexec/QtWebEngineProcess 
  *  create-default-gl-context  
  *  disable-features DnsOverHttpsUpgrade,ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-setuid-sandbox  
  *  disable-speech-api  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-threaded-compositing  
  *  in-process-gpu  
  *  use-gl desktop 
```

I have unset the variable QT_QPA_PLATFORM.

**Comment by muihlinn** [comment]
That's what I get:

- About:
![image](https://user-images.githubusercontent.com/7160350/235899468-54cb0ae5-3912-41b7-aa3a-632df84196ff.png)

- If I try to install libqt6webenginecore6-bin `apt`returns this 
```
libqt6webenginecore6-bin is already the newest version (6.2.4+dfsg-6ubuntu1).
```
- console output: 
``` 
13:13:50.037 [INFO    ] [0x7f41fa3be480]                              [qt.webenginecontext] => 

GLImplementation: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.6
Using Default SG Backend: yes
Using Software Dynamic GL: no
Using Angle: no

Init Parameters:
  *  application-name QLog 
  *  browser-subprocess-path /usr/lib/qt6/libexec/QtWebEngineProcess 
  *  create-default-gl-context  
  *  disable-features DnsOverHttpsUpgrade,ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-setuid-sandbox  
  *  disable-speech-api  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-threaded-compositing  
  *  in-process-gpu  
  *  use-gl desktop 
 [(null):(null):0] 
13:13:50.187 [WARNING ] [0x7f41fa3be480]                                    [             ] => Ignored NaN, Inf, or -Inf value. [(null):(null):0] 
13:13:50.187 [WARNING ] [0x7f41fa3be480]                                    [             ] => Attempting to set invalid range for value axis: [ nan  -  nan ] [(null):(null):0] 
```

There are some slight differences like :  
- hamlib version (I have some issues with one of my rigs - like if I had more than two :D) I'm trying to fix some stuff for it, so I use a recent copy from repo. 
- PictureInPicture  appears as disable-features
- Surface version is 4.6 instead 4.3

I don't know about QT development, I guess that such variable is disabled in QLog.pro or in the makefile, anyway I'm using X11, not wayland.

**Comment by foldynl** [comment]
I'm sorry, but at the moment I'm not able to tell you what could be the root cause. I did a few tests, but I always got the map. What I tested
1) under XOrg
2) under wayland

The difference may be that I run it in a Virtual Machine and OpenGL render is virtual. 

Please try to look at Logbook->statistics. Set Graph to "Show on Map" Do you see the map there?
![image](https://user-images.githubusercontent.com/942769/235936051-e67d121a-9b93-44ae-ae0b-45d6bf3e1ca4.png)

HAMLib version has no impact on the map.

**Comment by muihlinn** [comment]
No,  I get the white rectangle.

![image](https://user-images.githubusercontent.com/7160350/235939493-9d3468b1-b887-4f38-85cc-b826a0a6af04.png)

or anywhere else where the map should appear, as in... 

![image](https://user-images.githubusercontent.com/7160350/235939310-4440da9d-c94f-4af7-958a-c3ed3a858d9f.png)

also I get this on the console, but I have no idea of what it is related to. 

    [(null):(null):0] 
    16:05:30.886 [WARNING ] [0x7f95a3aab480]                                    [             ] => Ignored NaN, Inf, or -Inf value. [(null):(null):0] 
    16:05:30.886 [WARNING ] [0x7f95a3aab480]                                    [             ] => Attempting to set invalid range for value axis: [ nan  -  nan ] [(null):(null):0] 

Apart this all I've used works as intended.

**Comment by foldynl** [comment]
Are you running XOrg or Wayland ?

It seems that that is something wrong in Ubuntu's QT6 (version 6.2.4). I have tested the QLog with QT6 on the real hardware. 

1) HW Intel Graphics card  - everything is OK - under wayland and XOrg
2) HW GeForce RTX 3050 Ti Mobile - Wayland is OK, but under emulation of XOrg. If wayland module is enabled  then QLog crashes
3) HW GeForce RTX 3050 Ti Mobile - QLog is not running under XOrg - there is an issue with my keyboard setting - I don't know why, I have to googling.
4) HW AMD graphics card, Windows 10, QT 6.4.2 - without any issue.

do you have any specific requirements to use qt6 under ubuntu 22.04? To be honest, early versions of QT6 were quite buggy. I use QT6 for QLog only under Windows and MacOS because they prepare the latest versions for these Systems.

**Reply by muihlinn** [reply]
I'm using xorg. I've moved off wayland (default on 22.4) because it's promising but not really delivering for most people with quite a few customized things on xorg unsupported on wayland.  On QT6, probably I had a reason two or three months back, but now isn't a problem going back to QT5.  I'll try and will tell here.

**Reply by foldynl** [reply]
The issue under XOrg is explained. Just summarize the issue. I am getting following error with QT 6.2.4 under XOrg.

xkbcommon: ERROR: Couldn't process include statement for 'us(qwerty)'
xkbcommon: ERROR: Abandoning symbols file "(unnamed)"
xkbcommon: ERROR: Failed to compile xkb_symbols
xkbcommon: ERROR: Failed to compile keymap

It is a confirmed issue in QT6.2.3 and QT6.2.4; fixed in QT 6.2.5 (https://bugreports.qt.io/browse/QTBUG-101201). 

So I guess that it is not a good time to run QLog under QT 6.2.x 

Better situation can be under the new Ubuntu 23.04 where QT 6.4.2 is present .

**Reply by muihlinn** [reply]
Understood. 
Unfortunately compiling it with QT5 packages leaded to a segfault in the executable, and the package does that too, as right now I don't have time to dig what's going on, I'd use qlog without online maps in this machine.

**Reply by foldynl** [reply]
I'm sorry that I have such problems with the QLog. I try to test QLog as much as possible, but it is not possible to try all variants - especially under Linux because Linux users are used their own customization, and Linux distributions contains different version of libraries, sometime very old versions.

**Reply by muihlinn** [reply]
yes, I'm not really happy with this ubuntu LTS, had to fix a boatload of things, and I still have a TODO list of things that I'd change.  

Don't worry, QLog is fine to me, I guess that those segfaults are things related to libraries racing or incorrectly linked dynamically in my system, but now I don't have the time to troubleshoot that, so no online maps are a minor annoyance which  I can live with.
